### PR TITLE
Add a glossary to the FAQ

### PR DESCRIPTION
**Pull request by danintel**
- Add Avalon glossary section
- Minor formatting fixes

Signed-off-by: danintel <daniel.anderson@intel.com>

### CODE REVIEW

**Comment by danintel** [comment]
The rendered glossary, which is easier to read, is at:
https://github.com/danintel/avalon/blob/glossary/docs/FAQ.rst#glossary

File: docs/FAQ.rst [hunk 1]
```diff
@@ -341,4 +343,347 @@ If you are behind a corporate proxy, make sure you have ``proxy type`` and
...
 
+Glossary
+========
+
+AES-GCM 256
+    Avalon's authenticated encryption algorithm, with a 96b IV
+    and 128b tag. Encrypts data within a work order request or response.
+    Also used to encrypt a request digest and custom data encryption keys
+
+application type ID
```
Comment (Ram-srini) [inline docs/FAQ.rst]: s/application/Application/
Can we capital character for first letter(kind of title case)

File: docs/FAQ.rst [hunk 2]
```diff
@@ -341,4 +343,347 @@ If you are behind a corporate proxy, make sure you have ``proxy type`` and
...
+    Kubernetes container platform
+
+KECCACK-256
+    One of Avalon's digest algorithms. Used for work order
+    requests and responses or Ethereum raw transaction packet bytes. Bitcoin
+    and other blockchains used an early form of Keccak, "submitted version
+    3", before Keccak was standardized to SHA-3 (FIPS-202) and Keccak has
+    minor variations from SHA-3
+
+kv
```
Comment (Ram-srini) [inline docs/FAQ.rst]: s/kv/KV

File: docs/FAQ.rst (hunk 1 above)
Comment (danintel) [reply docs/FAQ.rst]: Done.

File: docs/FAQ.rst (hunk 2 above)
Comment (danintel) [reply docs/FAQ.rst]: Done.

**Pull request merged**